### DISCUSSION TITLE
Install Requirements

### DISCUSSION THREAD

**Discussion by ts2021** [Arize-ai/phoenix#2160] [General]
Hello, What are the installation requirements (Visual Studio version, python version, cython, etc.) for installing Arize-phoenix? I was unable to build on Windows 10 using pip, conda or from a wget download. hdbscan failing was the error.

**Comment by RogerHYang** [comment]
Hi @ts2021. Would you mind posting the error message here for us to take a look?

**Reply by ts2021** [reply]
The error is below. I need to update my Windows 10 SDK it seems. 

pip install arize-phoenix
Building wheels for collected packages: hdbscan
  Building wheel for hdbscan (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Building wheel for hdbscan (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [32 lines of output]
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build\lib.win-amd64-cpython-311
      creating build\lib.win-amd64-cpython-311\hdbscan
      copying hdbscan\flat.py -> build\lib.win-amd64-cpython-311\hdbscan
      copying hdbscan\hdbscan_.py -> build\lib.win-amd64-cpython-311\hdbscan
      copying hdbscan\plots.py -> build\lib.win-amd64-cpython-311\hdbscan
      copying hdbscan\prediction.py -> build\lib.win-amd64-cpython-311\hdbscan
      copying hdbscan\robust_single_linkage_.py -> build\lib.win-amd64-cpython-311\hdbscan
      copying hdbscan\validity.py -> build\lib.win-amd64-cpython-311\hdbscan
      copying hdbscan\__init__.py -> build\lib.win-amd64-cpython-311\hdbscan
      creating build\lib.win-amd64-cpython-311\hdbscan\tests
      copying hdbscan\tests\test_flat.py -> build\lib.win-amd64-cpython-311\hdbscan\tests
      copying hdbscan\tests\test_hdbscan.py -> build\lib.win-amd64-cpython-311\hdbscan\tests
      copying hdbscan\tests\test_prediction_utils.py -> build\lib.win-amd64-cpython-311\hdbscan\tests
      copying hdbscan\tests\test_rsl.py -> build\lib.win-amd64-cpython-311\hdbscan\tests
      copying hdbscan\tests\__init__.py -> build\lib.win-amd64-cpython-311\hdbscan\tests
      running build_ext
      C:\Users\***\AppData\Roaming\Python\Python311\site-packages\Cython\Compiler\Main.py:381: FutureWarning: Cython directive 'language_level' not set, using '3str' for now (Py3). This has changed
from earlier releases! File: C:\Users\***\AppData\Local\Temp\1\pip-install-oqud1i15\hdbscan_4365f82b88484ea988f178fcfa815f3c\hdbscan\_hdbscan_tree.pyx
        tree = Parsing.p_module(s, pxd, full_module_name)
      Compiling hdbscan/_hdbscan_tree.pyx because it changed.
      [1/1] Cythonizing hdbscan/_hdbscan_tree.pyx
      building 'hdbscan._hdbscan_tree' extension
      creating build\temp.win-amd64-cpython-311
      creating build\temp.win-amd64-cpython-311\Release
      creating build\temp.win-amd64-cpython-311\Release\hdbscan
      "C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\VC\Tools\MSVC\14.16.27023\bin\HostX86\x64\cl.exe" /c /nologo /O2 /W3 /GL /DNDEBUG /MD -IC:\Users\***\AppData\Local\miniconda3\
include -IC:\Users\***\AppData\Local\miniconda3\Include -IC:\Users\***\AppData\Roaming\Python\Python311\site-packages\numpy\core\include "-IC:\Program Files (x86)\Microsoft Visual Studio\2017\Pro
fessional\VC\Tools\MSVC\14.16.27023\ATLMFC\include" "-IC:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\VC\Tools\MSVC\14.16.27023\include" "-IC:\Program Files (x86)\Windows Kits\NETFXSD
K\4.6.1\include\um" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.10240.0\ucrt" /Tchdbscan/_hdbscan_tree.c /Fobuild\temp.win-amd64-cpython-311\Release\hdbscan/_hdbscan_tree.obj
      _hdbscan_tree.c
      c:\users\***\appdata\local\miniconda3\include\pyconfig.h(207): fatal error C1083: Cannot open include file: 'basetsd.h': No such file or directory
      error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\Professional\\VC\\Tools\\MSVC\\14.16.27023\\bin\\HostX86\\x64\\cl.exe' failed with exit code 2
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for hdbscan
Failed to build hdbscan
ERROR: Could not build wheels for hdbscan, which is required to install pyproject.toml-based projects

**Reply by RogerHYang** [reply]
Thanks. I did some google search and people seem to recommend conda for hdbscan in general.
```
conda install -c conda-forge hdbscan
```
Phoenix is available on conda as well, did that also fail to install for you?

**Reply by ts2021** [reply]
The conda install worked. Thanks

**Reply by RogerHYang** [reply]
Cool. Glad that worked out. Thanks for trying Phoenix!

**Comment by mikeldking** [comment]
Glad this is resolved! Let us know if you have any more windows issues.
